pkg/cmd/cli/uninstall/uninstall.go

```diff
@@ -19,96 +19,158 @@ package uninstall
+			fmt.Printf("Velero installation namespace %q is terminating.\n", namespace)
+		} else {
+			err = client.CoreV1().Namespaces().Delete(ctx, ns.Name, metav1.DeleteOptions{})
+			if err != nil {
+				errs = append(errs, errors.WithStack(err))
+			}
+		}
 	}
-	veleroLabels := labels.FormatLabels(install.Labels())
 
+	// rolebinding
+	crb := install.ClusterRoleBinding(namespace)
+	if err := client.RbacV1().ClusterRoleBindings().Delete(ctx, crb.Name, metav1.DeleteOptions{}); err != nil {
+		if apierrors.IsNotFound(err) {
+			fmt.Printf("Velero installation clusterrolebinding %q does not exist, skipping.\n", crb.Name)
+		} else {
+			errs = append(errs, errors.WithStack(err))
+		}
+	}
+
+	// CRDs
+	veleroLabels := labels.FormatLabels(install.Labels())
 	crds, err := extensionsClient.ApiextensionsV1().CustomResourceDefinitions().List(ctx, metav1.ListOptions{
 		LabelSelector: veleroLabels,
 	})
 	if err != nil {
-		return errors.WithMessagef(err, "Uninstall failed listing Velero crds")
+		errs = append(errs, errors.WithStack(err))
 	}
-	for _, removeCRD := range crds.Items {
-		err = extensionsClient.ApiextensionsV1().CustomResourceDefinitions().Delete(ctx, removeCRD.ObjectMeta.Name, metav1.DeleteOptions{})
-		if err != nil {
-			return errors.WithMessagef(err, "Uninstall failed removing CRD %s", removeCRD.ObjectMeta.Name)
+	if len(crds.Items) == 0 {
+		fmt.Print("Velero CRDs do not exist, skipping.\n")
+	} else {
+		for _, removeCRD := range crds.Items {
+			if err = extensionsClient.ApiextensionsV1().CustomResourceDefinitions().Delete(ctx, removeCRD.ObjectMeta.Name, metav1.DeleteOptions{}); err != nil {
+				err2 := errors.WithMessagef(err, "Uninstall failed removing CRD %s", removeCRD.ObjectMeta.Name)
+				errs = append(errs, errors.WithStack(err2))
+			}
 		}
 	}
 
-	fmt.Println("Uninstalled Velero")
-	return nil
-}
+	if waitToTerminate && len(ns.Name) != 0 {
+		fmt.Println("Waiting for Velero uninstall to complete. You may safely press ctrl-c to stop waiting - uninstall will continue in the background.")
 
-func DeleteNamespace(ctx context.Context, client *kubernetes.Clientset, namespace string) error {
-	err := client.CoreV1().Namespaces().Delete(ctx, namespace, metav1.DeleteOptions{})
-	if err != nil {
-		return errors.WithMessagef(err, "Delete namespace failed removing namespace %s", namespace)
-	}
-	return wait.Poll(1*time.Second, 3*time.Minute, func() (bool, error) {
-		_, err := client.CoreV1().Namespaces().Get(ctx, namespace, metav1.GetOptions{})
-		if err != nil {
-			// Commented this out because not sure if removing this is okay
-			// Printing this on Uninstall will lead to confusion
-			// fmt.Printf("Namespaces.Get after delete return err %v\n", err)
-			return true, nil // Assume any error means the delete was successful
+		ctx, cancel := context.WithCancel(ctx)
+		defer cancel()
+
+		checkFunc := func() {
+			nsUpdated, _ := client.CoreV1().Namespaces().Get(ctx, namespace, metav1.GetOptions{})
```

Comment:
```suggestion
			nsUpdated, err := client.CoreV1().Namespaces().Get(ctx, namespace, metav1.GetOptions{})
```

Reply:
We should also suppress `NotFound` error here.

Reply:
The variable `nsUpdated` makes me feel like we are performing an update operation. Can this just be `ns`?